### PR TITLE
BOAC-2709: Adds API for add/delete dept memberships

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2709

### CODE REVIEW

File: boac/api/user_controller.py [hunk 1]
```diff
@@ -59,6 +61,35 @@ def user_by_uid(uid):
     return tolerant_jsonify(calnet.get_calnet_user_for_uid(app, uid))
 
 
+@app.route('/api/user/dept_membership', methods=['POST'])
```
Comment (johncrossman) [inline boac/api/user_controller.py]: @lyttam - Our API paths tend to have a verb included. Eg, `/api/user/dept_membership/add`.  More readable?

File: boac/api/user_controller.py [hunk 2]
```diff
@@ -59,6 +61,35 @@ def user_by_uid(uid):
...
+        university_dept=dept,
+        authorized_user=user,
+        is_advisor=params.get('is_advisor', None),
+        is_director=params.get('is_director', None),
+        automate_membership=params.get('automate_membership', None),
+    )
+    return tolerant_jsonify(membership.to_api_json())
+
+
+@app.route('/api/user/dept_membership/<university_dept_id>/<authorized_user_id>', methods=['DELETE'])
```
Comment (johncrossman) [inline boac/api/user_controller.py]: @lyttam `/api/user/<authorized_user_id>/dept_membership/<university_dept_id>/delete`?

File: boac/api/user_controller.py (hunk 1 above)
Comment (lyttam) [reply boac/api/user_controller.py]: Good suggestion, I will make those changes.

File: boac/api/user_controller.py [hunk 3]
```diff
@@ -59,6 +61,35 @@ def user_by_uid(uid):
...
+def add_university_dept_membership():
+    params = request.get_json() or {}
+    dept = UniversityDept.find_by_dept_code(params.get('dept_code', None))
+    user = AuthorizedUser.find_by_uid(params.get('uid', None))
+    membership = UniversityDeptMember.create_membership(
+        university_dept=dept,
+        authorized_user=user,
+        is_advisor=params.get('is_advisor', None),
+        is_director=params.get('is_director', None),
+        automate_membership=params.get('automate_membership', None),
```
Comment (johncrossman) [inline boac/api/user_controller.py]: @lyttam  I suggest camelCase for params.

File: boac/models/university_dept_member.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        std_commit()
+        return True
+
+    def to_api_json(self):
+        return {
+            'university_dept_id': self.university_dept_id,
+            'authorized_user_id': self.authorized_user_id,
+            'is_advisor': self.is_advisor,
+            'is_director': self.is_director,
+            'automate_membership': self.automate_membership,
```
Comment (johncrossman) [inline boac/models/university_dept_member.py]: @lyttam - camelCase for `api_json` keys, methinks.

**Comment by johncrossman** [comment]
Thank you @lyttam  I'll merge once your changes are in.

**Pull request merged**